extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx

```diff
@@ -613,8 +732,18 @@ const createHistorySections = async (
     publicKey: string;
     networkDetails: NetworkDetails;
   }) => Promise<TokenDetailsResponse | Error>,
-) =>
-  await operations.reduce(
+  homeDomains: { [assetIssuer: string]: string | null },
+) => {
+  /* 
+    To prevent multiple requests for home domains as we build each row, 
+    we iterate through the operations and collect the asset issuers that need home domains in a single request.
+  */
+  const fetchedHomeDomains = await getHomeDomainsForOperations(
```

Comment:
When building account history rows, we used to iterate over operations while building rows and (if needed) make a request for home domains along the way in order to fetch the icon url. This led to a scenario where we could make up to 100 separate requests for a home domain for account history.